packages/Popover/Trigger.tsx

```diff
@@ -1,55 +1,73 @@
+        popoverRef.current?.removeEventListener('mouseenter', showPopover)
+        // add listeners on mouseleave
+        disclosureRef.current?.addEventListener('mouseleave', hidePopover)
+        popoverRef.current?.addEventListener('mouseleave', hidePopover)
+        // show popover
+        rest.show()
+      }
     }
-  }
 
-  const hidePopover: () => void = () => {
-    if (hoverable) {
-      // remove listeners on mouseleave
-      disclosureRef.current?.removeEventListener('mouseleave', hidePopover)
-      popoverRef.current?.removeEventListener('mouseleave', hidePopover)
-      // add listeners on mouseenter
-      disclosureRef.current?.addEventListener('mouseenter', showPopover)
-      popoverRef.current?.addEventListener('mouseenter', showPopover)
-      // hide popover
-      rest.hide()
+    const hidePopover: () => void = () => {
+      if (hoverable) {
+        // remove listeners on mouseleave
+        disclosureRef.current?.removeEventListener('mouseleave', hidePopover)
+        popoverRef.current?.removeEventListener('mouseleave', hidePopover)
+        // add listeners on mouseenter
+        disclosureRef.current?.addEventListener('mouseenter', showPopover)
+        popoverRef.current?.addEventListener('mouseenter', showPopover)
+        // hide popover
+        rest.hide()
+      }
     }
-  }
 
-  useLayoutEffect(() => {
-    const disclosure = disclosureRef.current
-    if (hoverable && disclosure) {
-      // add listeners on mount
-      disclosure.addEventListener('mouseenter', showPopover)
-      disclosure.addEventListener('mouseleave', hidePopover)
-      return () => {
-        // remove listeners on unmount
-        disclosure.removeEventListener('mouseenter', showPopover)
-        disclosure.removeEventListener('mouseleave', hidePopover)
+    useLayoutEffect(() => {
+      const disclosure = disclosureRef.current
+      if (hoverable && disclosure) {
+        // add listeners on mount
+        disclosure.addEventListener('mouseenter', showPopover)
+        disclosure.addEventListener('mouseleave', hidePopover)
+        return () => {
+          // remove listeners on unmount
+          disclosure.removeEventListener('mouseenter', showPopover)
+          disclosure.removeEventListener('mouseleave', hidePopover)
+        }
       }
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [disclosureRef])
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [disclosureRef])
 
-  return <PopoverDisclosure {...rest} ref={disclosureRef} />
-}
+    return (
+      <S.PopoverTrigger
+        {...rest}
+        forwardedAs={as}
+        ref={(triggerElement: HTMLButtonElement) => {
+          disclosureRef.current = triggerElement
+          rest.unstable_disclosureRef.current = triggerElement
+          if (typeof ref === 'function') {
+            ref(triggerElement)
+          } else if (ref?.current) {
+            ref.current = triggerElement
+          }
```

Comment:
possible de faire ça dans une function plus haut ?

Reply:
it's fixed :)